### PR TITLE
Upgrade OpenAI SDK to 2.14.0

### LINKED ISSUE #7760: Request to upgrade OpenAI library to 2.14.0 to fix bugs in OpenAI 2.13.0
### Description

# Summary
Two chained defects make the Responses API unusable against third-party OpenAI-compatible endpoints that emit `"status": null` on reasoning items:

1. **2.13.0**: `DeserializeReasoningResponseItem` lacks a null guard for the optional `status` property, so `"status": null` throws `ArgumentOutOfRangeException` and aborts the entire SSE stream.
2. **2.14.0**: the null guard was added, but the release also **removed the public (experimental) struct `OpenAI.Responses.GlobalMcpToolCallApprovalPolicy`**, which `Microsoft.Extensions.AI.OpenAI` 10.10.0 references from `OpenAIResponsesChatClient.ToResponseTool`. Any tool-carrying Responses call through the M.E.AI adapter now dies with `TypeLoadException` at JIT time. 

## Environment

- OpenAI SDK: 2.13.0 (also tested 2.14.0)
- Microsoft.Agents.AI.OpenAI 1.21.0 / Microsoft.Extensions.AI.OpenAI 10.10.0
- .NET 10

## Defect 1 — 2.13.0: no null guard on `status` (repro)

Third-party Responses-compatible backends may serialize the optional `status` field of a reasoning item as JSON `null` instead of omitting it (same class of report as https://github.com/openai/openai-dotnet/issues/1342). Since `status` is optional per spec, `null` should deserialize the same as "absent".

This issue has already been addressed https://github.com/openai/openai-dotnet/pull/1346 repaired.

Minimal repro — no network or API key required, uses the exact entry point the SSE pipeline uses (`AsyncSseUpdateCollection` → `ModelReaderWriter`):

```csharp
using System.ClientModel.Primitives;
using OpenAI.Responses;

var json = """
           {"type":"response.output_item.added","output_index":0,
            "item":{"type":"reasoning","id":"rs_1","status":null,"summary":[]}}
           """;

#pragma warning disable OPENAI001
var update = ModelReaderWriter.Read<StreamingResponseUpdate>(BinaryData.FromString(json));
#pragma warning restore OPENAI001
```

**Actual (2.13.0):**
```
System.ArgumentOutOfRangeException: Unknown ReasoningStatus value. (Parameter 'value')
   at OpenAI.Responses.ReasoningStatusExtensions.ToReasoningStatus(String value)
   at OpenAI.Responses.ReasoningResponseItem.DeserializeReasoningResponseItem(...)
```

**Expected:** `status: null` treated as absent (`ReasoningStatus? status = null`).

**Actual (2.14.0):** passes — decompilation confirms the guard was added:

```csharp
// 2.13.0 (generated) — no null check; GetString() on JSON null returns C# null
else if (item.NameEquals("status"))
{
    status = item.Value.GetString().ToReasoningStatus();  // throws
}

// 2.14.0 (generated)
else if (item.NameEquals("status"))
{
    if (item.Value.ValueKind != JsonValueKind.Null)
    {
        status = item.Value.GetString().ToReasoningStatus();
    }
}
```

## Defect 2 — 2.14.0: breaking removal of `GlobalMcpToolCallApprovalPolicy`

I tried importing OpenAI 2.14.0 package directly, but it caused this problem:

```
System.TypeLoadException: Could not load type 'OpenAI.Responses.GlobalMcpToolCallApprovalPolicy'
from assembly 'OpenAI, Version=2.14.0.0, Culture=neutral, PublicKeyToken=b4187f3e65366280'.
   at Microsoft.Extensions.AI.OpenAIResponsesChatClient.ToResponseTool(AITool tool, ChatOptions options, ToolSearchLookup toolSearchLookup)
   at Microsoft.Extensions.AI.OpenAIResponsesChatClient.AsCreateResponseOptions(...)
   at Microsoft.Extensions.AI.OpenAIResponsesChatClient.GetStreamingResponseAsync(...)
```

In 2.14.0, `GlobalMcpToolCallApprovalPolicy` has been renamed to `DefaultMcpToolCallApprovalPolicy`. The following is a screenshot of [openai-dotnet CHANGELOG](https://github.com/openai/openai-dotnet/blob/main/CHANGELOG.md).

Image

### Reproduction Steps

```csharp
using System.ClientModel.Primitives;
using OpenAI.Responses;

var json = """
           {"type":"response.output_item.added","output_index":0,
            "item":{"type":"reasoning","id":"rs_1","status":null,"summary":[]}}
           """;

#pragma warning disable OPENAI001
var update = ModelReaderWriter.Read<StreamingResponseUpdate>(BinaryData.FromString(json));
#pragma warning restore OPENAI001
```

### Expected behavior

Correct deserialization to obtain StreamResponseUpdate: `status: null` treated as absent (`ReasoningStatus? status = null`).

### Actual behavior

**2.13.0:**
```
System.ArgumentOutOfRangeException: Unknown ReasoningStatus value. (Parameter 'value')
   at OpenAI.Responses.ReasoningStatusExtensions.ToReasoningStatus(String value)
   at OpenAI.Responses.ReasoningResponseItem.DeserializeReasoningResponseItem(...)

Unhandled exception. System.ArgumentOutOfRangeException: Unknown ReasoningStatus value. (Parameter 'value')
   at OpenAI.Responses.ReasoningStatusExtensions.ToReasoningStatus(String value)
   at OpenAI.Responses.ReasoningResponseItem.DeserializeReasoningResponseItem...
```

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by zxyao145**
Fixes #7760

Bump the OpenAI package from 2.13.0 to 2.14.0. The new release renames GlobalMcpToolCallApprovalPolicy to DefaultMcpToolCallApprovalPolicy (and the GlobalPolicy property to DefaultPolicy), so update the MCP approval-policy mapping in OpenAIResponsesChatClient and the affected tests accordingly.

Also add a streaming regression test for OpenAI-compatible endpoints that serialize a reasoning item's optional "status" property as null instead of omitting it, which previously tore down the SSE stream with ArgumentOutOfRangeException ("Unknown ReasoningStatus value").
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/extensions/pull/7761)

### CODE REVIEW

**Review by wtgodbe — APPROVED** [review]

File: test/Libraries/Microsoft.Extensions.AI.OpenAI.Tests/OpenAIResponseClientTests.cs [hunk 1]
```diff
@@ -476,6 +476,112 @@ public async Task ReasoningTextDelta_Streaming()
         Assert.Equal(35, usage.Details.TotalTokenCount);
     }
 
+    [Fact]
+    public async Task ReasoningItemNullStatus_Streaming()
```
Comment (jozkee) [inline test/Libraries/Microsoft.Extensions.AI.OpenAI.Tests/OpenAIResponseClientTests.cs:480]: For future reference, unless we’re implementing a workaround in MEAI, I don’t think we should add a test for an OpenAI SDK bug.

**Review by jozkee — APPROVED** [review]
Thanks.

File: test/Libraries/Microsoft.Extensions.AI.OpenAI.Tests/OpenAIResponseClientTests.cs (hunk 1 above)
Comment (zxyao145) [reply test/Libraries/Microsoft.Extensions.AI.OpenAI.Tests/OpenAIResponseClientTests.cs]: Thanks for the clarification! I'll keep that in mind and only add tests for SDK bugs when we're implementing a workaround in MEAI.

**Comment by zxyao145** [comment]
Hi, this is my first PR to this repo. The extensions-ci seems to have failed due to **Windows_CodeCoverageResults already exists**, while the build and tests passed. 

Could someone please let me know how I can **rerun the failed CI**? Thanks!

image
